src/main/java/mServer/crawler/sender/ard/ArdVideoDeserializer.java

```diff
@@ -21,74 +23,122 @@
-                } else if (quality == Qualities.NORMAL) {
-                    // bei normaler Auflösung ist aus irgendeinem Grund der letzte
-                    // Eintrag immer der beste!
-                    urlMap.put(quality, urlInfos);
-                }
-            } else {
-                urlMap.put(quality, urlInfos);
+        switch(qualityDto.quality){
+          case "1":
+            // für SMALL die erste URL nehmen, ist immer die bessere
+            addUrl(urlMap, Qualities.SMALL, qualityDto.urls.get(0));
+            break;
+          case "2":
+            // für NORMAL die letzte URL nehmen, ist immer die bessere
+            addUrl(urlMap, Qualities.NORMAL, qualityDto.urls.get(qualityDto.urls.size()-1));
+            break;
+          case "3": 
+            if (qualityDto.width >= 1280 && qualityDto.height >= 720) {
+              addUrl(urlMap, Qualities.HD, getUrl(qualityDto, qualities.get(i+1)));
             }
-        }        
+            if (qualityDto.width == 0 && qualityDto.height == 0) {
+              String url = getUrl(qualityDto, qualities.get(i+1));
+              
+              // Die Prüfung auf den Dateinamen der URL filtert noch zusätzlich
+              // Filme heraus, die eine Auflösung von 960x... haben
+              if (!url.substring(url.lastIndexOf('/') + 1).startsWith("960")) {
+                addUrl(urlMap, Qualities.HD, url);
+              } else {
+                addUrl(urlMap, Qualities.NORMAL, url);
+              }
+            }
+            break;
+        }
+      }
     }
     
-    private UrlWidthHeightDTO deserializeUrlInfos(JsonObject stream) {
-        UrlWidthHeightDTO dto = new UrlWidthHeightDTO();
+    private void addUrl(Map<Qualities, String> urlMap, Qualities quality, String url) {
+      if (url != null && !url.isEmpty() && !urlMap.containsKey(quality)) {
+        urlMap.put(quality, url);
+      }
+    }
+    
+    private String getUrl(QualityUrlWidthHeightDTO quality, QualityUrlWidthHeightDTO lowerQuality) {
+      for (int i = 0; i < quality.urls.size(); i++) {
+        String url = quality.urls.get(i);
+        
+        if (!lowerQuality.urls.contains(url)) {
+          return url;
+        }
+      }
+      
+      return "";
+    }
+ 
+    private QualityUrlWidthHeightDTO deserializeUrlInfos(JsonObject stream) {
+        QualityUrlWidthHeightDTO dto = new QualityUrlWidthHeightDTO();
+        
+        dto.quality = stream.get(JSON_ELEMENT_QUALITY).getAsString();
         
-        // Url ermitteln
-        // Wenn es ein Array ist, den letzten Eintrag nehmen, der hat im Zweifel
-        // die bessere Auflösung!
         JsonElement streamElement = stream.get(JSON_ELEMENT_STREAM);
         if(streamElement.isJsonArray()) {
             JsonArray streamArray = streamElement.getAsJsonArray();
-            dto.url = streamArray.get(streamArray.size()-1).getAsString();
+            streamArray.forEach(arrayElement -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)